api/envoy/admin/v2alpha/config_dump.proto

```diff
@@ -34,14 +34,25 @@ message ConfigDump {
   repeated google.protobuf.Any configs = 1;
 }
 
+message UpdateFailureState {
+  // What the component configuration would have been if the update had succeeded.
+  google.protobuf.Any failed_state = 1;
```

Comment:
nit: I would probably call this `failed_configuration` but I don't feel that strongly about it.